pkg/alertmanager/alertmanager.go

```diff
@@ -300,12 +304,40 @@ func New(cfg *Config, reg *prometheus.Registry) (*Alertmanager, error) {
 		am.mux.Handle(a, http.NotFoundHandler())
 	}
 
+	// This route is an experimental Mimir extension to the receivers, API, so we put
+	// it under an additional prefix to avoid any confusion with upstream Alertmanager.
+	am.mux.Handle(
+		path.Join(am.cfg.ExternalURL.Path, "/experimental/api/v1/receivers"),
+		http.HandlerFunc(am.GetReceiversHandler))
+
```

Comment:
Open to suggestions for:
- What path to use, using the `/experimental` prefix seemed safest to avoid potential collisions with upstream routes.
- Whether this should be gated e.g. behind `-alertmanager.grafana-alertmanager-compatibility-enabled`